src/Event/Subscriber/BddSubscriber.php

```diff
@@ -203,10 +203,22 @@ public function afterStep(AfterStepTested $event)
             }
         }
 
-        $imageFileName = $this->takeScreenshot();
-        $httpTransactions = $this->getHttpTransactionEvents();
+        if (($resultCode == TestResult::FAILED) AND (Athena::settings()->get('bddScreenshotOnlyOnFailed')->orDefaultTo(true) == 'on')){
```

Comment:
In this situation is best to use booleans instead of 'on' or 'off' strings.

Notice that if the value is not present you default to 'true' however you then compare it to a string instead of a boolean.